Sources_v3/StreamChat/ChatClient.swift

```diff
@@ -69,19 +69,15 @@ public typealias ChatClient = _ChatClient<DefaultExtraData>
 ///
 public class _ChatClient<ExtraData: ExtraDataTypes> {
     /// The `UserId` of the currently logged in user.
-    @Atomic public var currentUserId: UserId = .anonymous
-    
+    @Atomic public var currentUserId: UserId?
+
     /// The current connection status of the client.
     ///
     /// To observe changes in the connection status, create an instance of `CurrentChatUserController`, and use it to receive
     /// callbacks when the connection status changes.
     ///
-    public var connectionStatus: ConnectionStatus {
-        ConnectionStatus(
-            webSocketConnectionState: webSocketClient?.connectionState ?? .disconnected(error: .ClientIsNotInActiveMode())
-        )
-    }
-    
+    public private(set) var connectionStatus: ConnectionStatus = .initialized
+
```

Comment:
There was a need to have some state the **ChatClient** stays in after instantiation until the `reconnectIfNeeded` is invoked.

Reply:
The client should start connection immediately unless it's not specified in config otherwise ⚠️

Reply:
fixed